app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardInstancePropertiesStep/PlanWizardInstancePropertiesStepReducer.js

```diff
@@ -23,6 +43,54 @@ export default (state = initialState, action) => {
         .set('errorTenantsWithAttributes', action.payload)
         .set('isFetchingTenantsWithAttributes', false)
         .set('isRejectedTenantsWithAttributes', true);
+    case `${QUERY_V2V_OSP_BEST_FIT_FLAVOR}_PENDING`:
+      return state.set('isFetchingBestFitFlavor', true).set('isRejectedBestFitFlavor', false);
+    case `${QUERY_V2V_OSP_BEST_FIT_FLAVOR}_FULFILLED`:
+      return state
+        .set('bestFitFlavors', action.payload.data.results)
+        .set('isFetchingBestFitFlavor', false)
+        .set('isRejectedBestFitFlavor', false)
+        .set('errorBestFitFlavor', null);
+    case `${QUERY_V2V_OSP_BEST_FIT_FLAVOR}_REJECTED`:
+      return state
+        .set('errorBestFitFlavor', action.payload)
+        .set('isFetchingBestFitFlavor', false)
+        .set('isRejectedBestFitFlavor', true);
+    case SET_V2V_BEST_FIT_FLAVORS_AND_DEFAULT_SECURITY_GROUPS: {
+      const vmBestFitFlavors = action.payload;
+      const instancePropertiesRowsUpdatedWithBestFlavor = [];
+      vmBestFitFlavors.forEach(vmFlavor => {
+        const existingInstancePropertiesRow = state.instancePropertiesRows.find(row => row.id === vmFlavor.vm_id);
+        const tenant = state.tenantsWithAttributes.find(
+          tenantsWithAttribute => tenantsWithAttribute.id === vmFlavor.tenant_id
+        );
+        const tenantFlavors = tenant.flavors;
+        const tenantSecurityGroups = tenant.security_groups;
+        const bestFitFlavorName = tenantFlavors.find(flavor => flavor.id === vmFlavor.flavor_id).name;
+        const defaultSecurityGroupName = 'default';
+        const defaultSecurityGroupId = tenantSecurityGroups.find(
+          securityGroup => securityGroup.name === defaultSecurityGroupName
+        ).id;
+        const rowUpdatedWithBestFlavor = {
+          ...existingInstancePropertiesRow,
+          osp_flavor: { name: bestFitFlavorName, id: vmFlavor.flavor_id },
+          osp_security_group: { name: defaultSecurityGroupName, id: defaultSecurityGroupId },
+          target_cluster_name: tenant.name
+        };
+        instancePropertiesRowsUpdatedWithBestFlavor.push(rowUpdatedWithBestFlavor);
+      });
+      return state
+        .set('isSettingSecurityGroupsAndBestFitFlavors', false)
+        .set('instancePropertiesRows', instancePropertiesRowsUpdatedWithBestFlavor);
+    }
+    case SET_V2V_INSTANCE_PROPERTIES_ROWS:
+      return state.set('instancePropertiesRows', action.payload);
+    case SET_UPDATED_INSTANCE_PROPERTIES_ROW_ON_STANDBY:
+      return state.set('updatedInstancePropertiesRowOnStandby', action.payload);
+    case PLAN_WIZARD_EXITED:
+      return initialState;
+    case PLAN_WIZARD_NEXT:
+      return state.set('recalculateBestFitFlavorAndSecurityGroup', action.payload === stepIDs.vmStep);
```

Comment:
This will fire when leaving the VM step even in the non-OSP case. That's probably fine, just thinking it through..